lms/djangoapps/certificates/models.py

```diff
@@ -294,9 +294,13 @@ def get_certificate_generation_candidates(self):
                 [CertificateStatuses.readable_statuses.get(status, "") for status in statuses]
             )
 
-        # If statuses_to_regenerate is not present in task_input then, certificate generation task was run to
+        # If students is present in task_input then, certificate generation task was run to
         # generate certificates for white listed students
-        return "for exceptions"
+        if 'students' in task_input_json:
+            return "for exceptions"
+
+        # If task input is empty dict ({}), it means certificates were generated for all learners
+        return "All learners"
```

Comment:
one return statement. 
`return "for exceptions" if 'students' in task_input_json else "All learners"`